src/shared/components/ChannelCover/ChannelCover.tsx

```diff
@@ -0,0 +1,70 @@
+import { BackgroundPattern } from '@/components'
+import { transitions } from '@/shared/theme'
+import React from 'react'
+import { CSSTransition, TransitionGroup } from 'react-transition-group'
+import Icon from '../Icon'
+import {
+  CoverImage,
+  EditableControls,
+  EditCoverButton,
+  EditIconWrapper,
+  EditButtonMessage,
+  Media,
+  MediaWrapper,
+  RemoveCoverButton,
+  CoverWrapper,
+} from './ChannelCover.style'
+
+export type ChannelCoverProps = {
+  coverPhotoUrl?: string | null
+  editable?: boolean
+  onCoverEditClick?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
+  onCoverRemoveClick?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
+}
+
+const ChannelCover: React.FC<ChannelCoverProps> = ({
+  coverPhotoUrl,
+  editable,
+  onCoverEditClick,
+  onCoverRemoveClick,
+}) => {
+  return (
+    <CoverWrapper>
+      <MediaWrapper>
+        {editable && (
+          <EditableControls>
+            <EditCoverButton onClick={onCoverEditClick}>
+              <EditIconWrapper>
+                <Icon className="desktop" name="camera" />
```

Comment:
Let's try to avoid using `className` like this. This creates a coupling that cannot be typechecked. Instead, we can do a `StyledIcon` that will accept a prop